pkg/parser/parser.go

```diff
@@ -133,17 +130,26 @@ func (p *PackageParser) Parse(_ context.Context, reader io.ReadCloser) (*Package
 	return pkg, nil
 }
 
-// isWhiteSpace determines whether the passed in bytes are all unicode white
-// space.
-func isWhiteSpace(bytes []byte) bool {
-	empty := true
-	for _, b := range bytes {
-		if !unicode.IsSpace(rune(b)) {
-			empty = false
-			break
+// cleanYAML cleans up YAML by removing empty and commented out lines which
+// cause issues with decoding.
+func cleanYAML(y []byte) []byte {
+	lines := []string{}
+	for _, line := range strings.Split(string(y), "\n") {
+		trimmed := strings.TrimSpace(line)
+		if trimmed == "" || strings.HasPrefix(trimmed, "#") {
+			continue
+		}
+		lines = append(lines, line)
+	}
+	// If all left are just separators and/or terminators, then there is no
+	// content.
+	for _, c := range lines {
+		trimmed := strings.TrimSpace(c)
```

Comment:
Do we need this TrimSpace here, given we already trimmed all lines above?

Reply:
In the first loop, we trim to check but don't add it to the final lines array because that'd shift all lines in the YAML to left, hence break it. So `lines` always has the original content. Maybe I should rename the function so it doesn't imply that - the only thing it does is to return empty when there is no content.

Reply:
In the first loop, the empty and commented lines are excluded but the ones that are included are untouched so we still need to trim to for an exact match.